sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -459,12 +548,284 @@ edges.
+----
+|Constructs a scalar dynamic parameter object that can be registered with node
+arguments.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Preconditions:
+
+* This constructor is only available when the templated `dynamic_parameter_type`
+  is `dynamic_parameter_type::scalar`.
+
+Parameters:
+
+* `size` - Size in bytes of the scalar object.
+
+* `label` - An optional string label which can be used to help identify the
+parameter which has been registered as dynamic.
+
+|
+[source,c++]
+----
+const std::string& get_label() const;
+----
+|Returns the label associated with the dynamic parameter.
+
+|
+[source,c++]
+----
+size_t get_size() const;
+----
+|Returns the size associated with the scalar dynamic parameter.
+
+Preconditions:
+
+* This constructor is only available when the templated `dynamic_parameter_type`
+  is `dynamic_parameter_type::scalar`.
+
+|===
+
+==== Node Member Functions
+
+Table {counter: tableNumber}. Member functions of the `node` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+void register_dynamic_parameter(
+    void* param,
+    dynamic_parameter<dynamic_parameter_type::usm> dynamicParam) const;
 ----
+|Associate a USM dynamic parameter with USM kernel parameter of this node.
+All USM kernel arguments matching `param` are bound to `dynamicParam`.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `param` - USM pointer to used as a node argument.
+* `dynamicParam` - Dynamic parameter which is to be associated with the kernel
+   arguments matching `param`.
+
+Exceptions:
+
+* Throws with error code `invalid` if `param` does not match any arguments in
+  the node.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+template <typename T>
+void register_dynamic_parameter(
+    T param,
+    dynamic_parameter<dynamic_parameter_type::buffer> dynamicParam) const;
+----
+|Associate a buffer dynamic parameter with buffer accessor kernel parameter of
+this node. All buffer accessor kernel arguments matching `param` are bound to
+`dynamicParam`.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `param` - Buffer accessor used as a node argument.
+* `dynamicParam` - Dynamic parameter which is to be associated with the kernel
+   arguments matching `param`.
+
+Exceptions:
+
+* Throws with error code `invalid` if `param` does not match any arguments in
+  the node.
+
+* Throws with error code `invalid` if `param` is not a buffer accessor.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+template <typename T>
+void register_dynamic_parameter(
+    T param,
+    dynamic_parameter<dynamic_parameter_type::scalar> dynamicParam) const;
```

Comment:
what will happen if a kernel has two different arguments with the same data type, and they happen to be the same value when the node is created.

Reply:
I've put this in the description below 

> All scalar kernel arguments matching `param` are bound to `dynamicParam`.

I couldn't think of a way to distinguish between scalars given they can be copied during pass-by-value, so we can't compare the memory address.